source/base/hdf5.cc

```diff
@@ -563,10 +556,10 @@ namespace HDF5
     /*
      * Write scalar attribute.
      */
-    ret = H5Awrite(attr, t_type, &c_string_value);
+    const char *c_string_value = value.c_str();
+    ret                        = H5Awrite(attr, t_type, &c_string_value);
```

Comment:
The way HDF5 deals with variable length data is a little bit confusing. Most of the times H5Awrite and H5Dwrite will take a pointer to the data. But in the particular case of a variable length string, it take the address of a pointer because you can eventually write an array of variable length strings.

How about you clarify this with a comment? for example:

_In most of the cases H5Awrite and H5Dwrite take a pointer to the data. But in the particular case of a variable length string H5Awrite takes the address of the pointer of the string._

Reply:
I'm happy with the rest of the modifications :)

Reply:
Yes, good idea. I updated the PR.